### PR TITLE
ucommon 7.0.1

### PR DESCRIPTION
**Pull request by chenrui333**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
https://github.com/Homebrew/homebrew-core/actions/runs/10405354654/job/28816939428?pr=181279#step:3:2854

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  thread.cpp:29:10: fatal error: sys/sysctl.h: No such file or directory
     29 | #include <sys/sysctl.h>
        |          ^~~~~~~~~~~~~~
  compilation terminated.
```

File: Formula/u/ucommon.rb [hunk 1]
```diff
@@ -1,16 +1,16 @@
 class Ucommon < Formula
   desc "GNU C++ runtime library for threads, sockets, and parsing"
   homepage "https://www.gnu.org/software/commoncpp/"
-  url "https://ftp.gnu.org/gnu/commonc++/ucommon-7.0.0.tar.gz"
-  sha256 "6ac9f76c2af010f97e916e4bae1cece341dc64ca28e3881ff4ddc3bc334060d7"
+  url "https://ftp.debian.org/debian/pool/main/u/ucommon/ucommon_7.0.1.orig.tar.gz"
```
Comment (carlocab) [inline Formula/u/ucommon.rb:4]: No tarball at gnu.org?

**Review by carlocab — APPROVED** [review]

File: Formula/u/ucommon.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/u/ucommon.rb:4]: yeah, a bit weird, but this is the 7.0.1 release thread, https://lists.gnu.org/archive/html/bug-commoncpp/2020-09/msg00001.html

Comment (chenrui333) [reply Formula/u/ucommon.rb:4]: I can leave a note there and ask why the gnu tarball missing.

Comment (carlocab) [reply Formula/u/ucommon.rb:4]: Sounds good!

**Review by iMichka — APPROVED** [review]

File: Formula/u/ucommon.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/u/ucommon.rb:4]: looks like no reply yet, https://lists.gnu.org/archive/html/bug-commoncpp/2024-08/msg00000.html :(

**Review by carlocab — APPROVED** [review]

**Pull request merged**